packages/cli/src/helpers/configExtractor.ts

```diff
@@ -0,0 +1,100 @@
+import Dotenv from 'dotenv';
+import _ from 'lodash';
+import { Cfgu, Config } from '@configu/sdk';
+
+type ConfigFormat =
+  | 'JSON'
+  | 'CompactJSON'
+  | 'YAML'
+  | 'Dotenv'
+  | 'KubernetesConfigMap'
+  | 'HelmValues'
+  | 'TerraformTfvars'
+  | 'TOML';
+
+const CONFIG_FORMAT_EXTENSION: Record<ConfigFormat, string> = {
+  JSON: 'json',
+  CompactJSON: 'json',
+  YAML: 'yaml',
+  Dotenv: 'env',
+  KubernetesConfigMap: 'yaml',
+  HelmValues: 'yaml',
+  TerraformTfvars: 'tfvars',
+  TOML: 'toml',
+};
+
+type ConfigExtractorFormat = Extract<ConfigFormat, 'JSON' | 'Dotenv'>;
+
+type ExtractorParameters = { content: string };
+type ExtractorFunction = (params: ExtractorParameters) => Record<string, string>;
+
+const FORMAT_TO_EXTENSION_DICT: Record<ConfigExtractorFormat, string> = _.pick(CONFIG_FORMAT_EXTENSION, [
+  'JSON',
+  'Dotenv',
+]);
+
+const FILE_CONTENT_FORMAT_ERROR =
+  'file content is not in a valid format. for more info on supported formats read here: https://configu.com/docs';
+const NOT_FLAT_JSON_ERROR = 'only flat .json files are supported. read more here: https://configu.com/docs';
+
+const CONFIG_EXTRACTORS: Record<ConfigExtractorFormat, ExtractorFunction> = {
+  Dotenv: ({ content }) => Dotenv.parse(content),
```

Comment:
we should use the new integrations

Reply:
The integrations are not relevant, they are for stringification while this logic is for parsing, two different purposes.

Reply:
Maybe we'll need another task to convert the usage here to the adaptive expressions, but that's another task.